developer_docs/remote_execution_dev_setup.asciidoc

```diff
@@ -0,0 +1,133 @@
+....
+
+Install plugin dependencies & migrate database
+[source, bash]
+....
+bundle install
+npm install
+
+bundle exec rails db:migrate
+....
+
+[[smart-proxy]]
+== Smart Proxy
+
+Add https://github.com/theforeman/smart_proxy_dynflow[smart_proxy_dynflow] and  https://github.com/theforeman/smart_proxy_remote_execution_ssh[smart_proxy_remote_execution_ssh] gems
+
+[source, bash]
+....
+
+# /path/to/smart-proxy/bundler.d/remote_execution.local.rb
+
+gem 'smart_proxy_dynflow', git: 'https://github.com/theforeman/smart_proxy_dynflow.git'
+gem 'smart_proxy_remote_execution_ssh', git: 'https://github.com/theforeman/smart_proxy_remote_execution_ssh.git'
+....
+
+Install dependencies
+[source, bash]
+....
+bundle install
+....
+
+Enable features
+
+[source, yaml]
+....
+# config/settings.d/dynflow.yml 
+---
+:enabled: true
+
+# config/settings.d/remote_execution_ssh.yml
+---
+:enabled: true
+....
+
+Run proxy and check that features have been enabled
+[source, bash]
+....
+2022-08-17T12:26:52  [I] Successfully initialized 'dynflow'
+2022-08-17T12:26:52  [I] Successfully initialized 'ssh'
+2022-08-17T12:26:52  [I] Successfully initialized 'foreman_proxy'
+....
+
+In Foreman on the detail of Smart Proxy click refresh and verify that `Script` (or `SSH`) and `Dynflow` features have been succesfully loaded.
+
+[[run]]
+== Run all servicies
+[source, bash]
+....
+bundle exec foreman start
+bundle exec rake dynflow:executor
+bundle exec bin/smart-proxy
+....
+
+[[configuring-foreman]]
+== Configuring Foreman
+
+In _Administer > Settings > Remote execution_
+
+* Set `Connect by IP` to `yes` so you don't have care about DNS
+* Optional: Set `default SSH password` that will be used for SSH connection. If you don't set the password, Remote Execution will try to authenticate with SSH keys
+
+[[try-if-it-works]]
+== Try if it works
+
+* In Foreman `console` create new host, with IP of the host from prerequisities
+
+[source, ruby]
+....
+Host.create(name: 'rex-test', managed: false, build: false, organization: Organization.find_by(name: 'Default Organization'), location: Location.find_by(name: 'Default Location'), ip: 'machine-ip4')
+....
+
+* In Foreman UI go to _Hosts > All Hosts_, select the `rex-test` and choose `Schedule Remote Job`
+* As command use for example `date` or `cat /etc/os-release`
+* Run job
+
+You can se the result for the host viewing the detail in the table at the bottom of job invocation page.
+
+[[troubleshooting]]
+== Troubleshooting
+=== Starting Smart Proxy - Missing SSH keys
+[source, bash]
+....
+Error while running proxy
+Error details for Couldn't enable 'ssh': <RuntimeError>: Ssh public key file /home/<your account>/.ssh/id_rsa_foreman_proxy doesn't exist.
+....
+
+Fix:
+[source, bash]
+....
+cd ~/.ssh
+ln -s id_rsa.pub ~/.ssh/id_rsa_foreman_proxy.pub
+ln -s id_rsa ~/.ssh/id_rsa_foreman_proxy
+....
+
+=== SSH debuging
+<ping-adam-here> how to enable SSH debugging and other stuff worth to mention
```

Comment:
`:ssh_log_level: debug` in `config/settings.d/remote_execution_ssh.yml`, `:log_level: DEBUG` in `config/settings.yml`, restart proxy.